pycbc/waveform/waveform.py

```diff
@@ -755,16 +820,24 @@ def imrphenomd_length_in_time(**kwds):
 cpu_fd["TaylorF2NL"] = nonlinear_tidal_spa
 
 for apx in copy.copy(_filter_time_lengths):
-    if apx in cpu_fd:
+    fd_apx = cpu_fd.keys()
+    td_apx = cpu_td.keys()
+
+    if (apx in td_apx) and (apx not in fd_apx):
+        # We can make a fd version of td approximants
+        cpu_fd[apx] = get_fd_waveform_from_td
+
+    if apx in fd_apx:
         # We can do interpolation for waveforms that have a time length
         apx_int = apx + '_INTERP'
         cpu_fd[apx_int] = get_interpolated_fd_waveform
         _filter_time_lengths[apx_int] = _filter_time_lengths[apx]
 
         # We can also make a td version of this
-        apx_int = apx + '_FD'
-        cpu_td[apx_int] = get_td_waveform_from_fd
-        _filter_time_lengths[apx_int] = _filter_time_lengths[apx]
+        # This will override any existing approximants with the same name
+        # (ex. IMRPhenomXX)
+        cpu_td[apx] = get_td_waveform_from_fd
```

Comment:
There's a potential problem here if someone decides to make their *TD* waveform available also in the frequency domain in lalsimulation!

I think my preference is to handle the Phenom cases specially, but it's not a strong preference.

Reply:
Indeed. I think we cross that bridge when we come to it though.